### ISSUE TITLE
Relationship-Multi-Attribute error on batch upsert

### ISSUE THREAD

**Issue by fbattello** [Fiware/context.Orion-LD#1246]
Hi,

Orion-LD correctly inserts the below entity using the **entities** endpoint, but returns an error when using the **batch upsert** endpoint.

```json
{
  "@context": [
    "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"
  ],
  "id": "urn:ngsi-ld:A:A1",
  "type": "A",
  "hasB": {
    "type": "Relationship",
    "object": "urn:ngsi-ld:B:B1"
  },
  "hasMultiD": [
    {
      "type": "Relationship",
      "object": "urn:ngsi-ld:D:D1",
      "datasetId": "urn:ngsi-ld:Relationship:1"
    },
    {
      "type": "Relationship",
      "object": "urn:ngsi-ld:D:D2",
      "datasetId": "urn:ngsi-ld:Relationship:2"
    }
  ]
}
```
Error Message :

> {"success":[],"errors":[{"entityId":"urn:ngsi-ld:A:A1","error":{"type":"https://uri.etsi.org/ngsi-ld/errors/BadRequestData","title":"attribute must be a JSON object","detail":"https://uri.etsi.org/ngsi-ld/default-context/hasMultiD","status":400}}]}

Orion-ld version : 

> 1.2.0-PRE-1117

**Comment by kzangeli** [comment]
Yeah, sorry about that.
Orion-LD doesn't support the datasetId (multi-attribute).

As a PoC, a tiny subset of the API was made "datasetId-aware" (2-3 operations) but, that's about it.
It's on the roadmap, of course, it's just not at the top (new NGSI-LD Registrations and Forwarding is - see issue #280)
And, forwarding is a major feature which is gonna take quite a while to finish (6 months?).
So, if you really need multi-attribute ... 
Really, what you seem to need is for a Relationship to support an array, and Orion-LD supports that - against the NGSI-LD API (long story - I'll tell you all about it if you're interested).

If you really need multi-attribute (which might take a year or so before it is fully implemented in Orion-LD), I'd have to invite you to have a look at Scorpio instead :(

(AFAIK, Stellio also doesn't fully support the datasetId)

**Comment by fbattello** [comment]
Ok, and thank you for the roadmap.
I better understand why it worked with a unitary operation but not a batch one.
I had to triple-check I built a valid request in batch mode :)
I need the multi-attribute as I develop the [ngsildclient](https://github.com/Orange-OpenSource/python-ngsild-client) Python library that implements _(a subset of)_ the NGSI-LD API on the client side.
For the moment I test mostly against Orion-LD, and it's fine as I have no strong need to go further with multi-attrs.
